parquet/src/test/java/org/apache/iceberg/parquet/TestDictionaryRowGroupFilter.java

```diff
@@ -156,22 +156,22 @@ public class TestDictionaryRowGroupFilter {
   private MessageType parquetSchema = null;
   private BlockMetaData rowGroupMetadata = null;
   private DictionaryPageReadStore dictionaryStore = null;
-  private final WriterVersion writerVersion;
-
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
-
-  @Parameterized.Parameters
-  public static List<WriterVersion> writerVersions() {
-    return Arrays.asList(PARQUET_1_0, PARQUET_2_0);
+  @Parameter
+  private WriterVersion writerVersion;
+
+  @Parameters(name = "writerVersion={0}")
+  public static Collection<WriterVersion> parameters() {
+    return Arrays.asList(
+            WriterVersion.PARQUET_1_0,
+            WriterVersion.PARQUET_2_0);
   }
+  
+  @TempDir
+  public Path temp;
 
-  public TestDictionaryRowGroupFilter(WriterVersion writerVersion) {
-    this.writerVersion = writerVersion;
-  }
-
-  @Before
+  @BeforeEach
   public void createInputFile() throws IOException {
-    File parquetFile = temp.newFile();
```

Comment:
I encounter the same case in `iceberg-core`. I step into the `newFile` function and find it create a temp file with prefix `junit`, using `File.createTempFile`. May be we can use the `File.createTempFile` directly in our code here?